com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ModuleDependencyChecker.java

```diff
@@ -82,4 +83,20 @@ private boolean areAllDependenciesAlreadyPresent(List<String> deployedAfter) {
     public Set<String> getAlreadyDeployedModules() {
         return modulesAlreadyDeployed;
     }
+
+    public Set<String> getNotProcessedModules() {
+        return modulesForDeployment.stream()
+                                   .filter(isProcessed().negate())
```

Comment:
Rather than using the `negate()` method, change `isProcessed()` to accept a parameter and return a boolean then negate that function